Automattic/wp-calypso
# Themes: Supply selectedSite prop to logged-in sheet

Addresses #4123

Will allow site-specific actions to be performed from the sheet. This PR also updates the navigation section names:
Details -> Overview
Documentation -> Setup
Support -> Support

**Before**
![screen shot 2016-05-05 at 12 07 32](https://cloud.githubusercontent.com/assets/7767559/15042249/58b7462c-12ba-11e6-856b-1b2da195094f.png)

**After**
![screen shot 2016-05-05 at 12 07 51](https://cloud.githubusercontent.com/assets/7767559/15042256/5e810c14-12ba-11e6-8dcc-d09d691b76d5.png)

The actions performed by the button remain a work-in-progress.

**To Test**
* Logged-in, with a site selected, pick a theme from alypso.localhost:3000/design and select _Details_
* Click around the various sections (Overview, Support)
Expected: the site slug should remain in the URL
Expected: themes without any support doc (e.g. Twenty Sixteen) should not show the _Setup_ tab

Extra:
* Logged-in, an invalid site slug on the URL should now redirect to the 'no-site' view
* Logged-out, any kind of site slug will redirect to the 'no-site' view
